core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -1977,6 +1977,12 @@ private void rewriteCase0(ExpressionNode node) {
             } else {
                 node.args.remove(paramCount - 1);
                 node.paramCount = paramCount - 1;
+
+                if (node.paramCount < 3) {
+                    node.rhs = node.args.get(0);
```

Comment:
ideally we should never have a case where paramCount < 3 has to be extracted from node.args. This is quite fundemantal for the compiler. Why is this assumption is broken here?

Reply:
right i see - i guess the core issue isn't fixed, which is incorrect case statement rewrite. This is errounously generated params (corrupt), I would not just unpack them

Reply:
it is a convention throughout the codebase (or it should be) that 2+ args go in the arg list, and < 3 args should always be in `left` and `right`. This is for consistency sake. Otherwise each code sight need to process expressions in paranoia mode.

I would prefer to add implicit null when we rewrite the case to make `case` compatible with PostgreSQL